### PR TITLE
Upgrade to Sequelize 6

### PR DESCRIPTION
**Pull request by kf6kjg**
BREAKING CHANGE!: The breakage comes from following Sequelize's lead and requiring a minimum node version of 10 and switching out Bluebird for native promises.

### CODE REVIEW

**Comment by kf6kjg** [comment]
Currently targeting the 6.0.0 beta release as that's what's available.

File: src/associations/belongs-to-many/belongs-to-many-association.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export class BelongsToManyAssociation extends BaseAssociation {
     const associatedClass = this.getAssociatedClass();
     const throughOptions = this.getThroughOptions(sequelize);
 
-    const throughModel = typeof throughOptions === 'object' ? throughOptions.model : undefined;
+    const throughModel = typeof throughOptions === 'object' && typeof throughOptions.model !== "string" ? throughOptions.model : undefined;
```
Comment (RobinBuschmann) [inline src/associations/belongs-to-many/belongs-to-many-association.ts:29]: Can you explain why this change is necessary?

File: src/sequelize/sequelize/sequelize-service.ts [hunk 2]
```diff
@@ -37,9 +38,9 @@ function getValidationOnlyOptions(options: SequelizeOptions): SequelizeOptions {
  * Determines models from value
  */
 export function getModels(
-  arg: Array<ModelCtor | string>,
+  arg: Array<ModelCtor<Model> | string>,
```
Comment (RobinBuschmann) [inline src/sequelize/sequelize/sequelize-service.ts:41]: Can you explain why you've add Model explicitly here? I'm asking since it is already set as default in ModelCtor definition

File: src/associations/belongs-to-many/belongs-to-many-association.ts (hunk 1 above)
Comment (kf6kjg) [reply src/associations/belongs-to-many/belongs-to-many-association.ts:29]: Because `throughOptions.model` has type `string | typeof Model`. The subsequent code isn't designed to handle the string case. Since my goals were to start with the minimal set of changes to make the code compatible with Sequelize 6 beta.6, this was the smallest fix to make the compiler and test suite happy.

File: src/sequelize/sequelize/sequelize-service.ts (hunk 2 above)
Comment (kf6kjg) [reply src/sequelize/sequelize/sequelize-service.ts:41]: Good catch: it and its siblings are legacy from another failed attempt that I missed in the cleanup phase.

**Comment by kf6kjg** [comment]
@RobinBuschmann Did I fully answer your questions and issues?

**Comment by brianschardt** [comment]
Hope you are all well! This would be amazing to be released, as I think a lot of people that can benefit from it.

**Comment by negezor** [comment]
Yesterday was the release of 6.1.0. It is worth checking for compatibility.

**Comment by kf6kjg** [comment]
At the moment I’m in crunch time through the end of the month. Is anyone else here able to see if this change works as is or not?

**Comment by dougmaitelli** [comment]
I see this PR is open for quite some time now and still reference Sequelize 6 beta, which is not beta anymore and this package is not currently working with Sequelize 6.

Is there any plan to fix this? Or is it supposed to work with Sequelize 6?

**Comment by kf6kjg** [comment]
As the author of this PR, yes the plan is to validate that it still works. But as I mentioned above I’m just a touch short on time at the moment. Would you be willing to build it and find out? It’d be very appreciated. :)

**Comment by dougmaitelli** [comment]
Hey @kf6kjg, I tried to test the code from your PR but it doesn't seem to work.
I can confirm the changes do fix the compilation errors with sequelize v6, but now application fails during start with the message:
``Error: Please install pg package manually``
However same project was working fine with v5 and pg is in fact installed.

**Comment by awais000** [comment]
I was just about to exclude typescript-sequelize from my project to migrate to seqeulize@6 and then came across to this pull request.  I'm going to test PR and see if it works for myself.

**Comment by awais000** [comment]
and unfortunately, it didn't work for me.  I gonna try to implement this on my own.

**Comment by brianschardt** [comment]
@awais000 i think a lot of people could benefit from this. if you do find a solution that would be dope.

**Comment by awais000** [comment]
@brianschardt I started working on it. I go release by release to implement and accommodate all the breaking changes. Almost done with 6.0.0 because of depreciation of blueBird some of the tests got disturbed.

**Comment by awais000** [comment]
#807 there you go @brianschardt I opened a pull request. Please do take a look at it.

**Comment by awais000** [comment]
#807 there you go @brianschardt I opened a pull request. Please do take a look at it.

**Comment by cyrilchapon** [comment]
How is this going ? :)

**Comment by Jinkwon** [comment]
is there any work in progress?

**Comment by kf6kjg** [comment]
I'm closing this as pretty much everything I did was done in the already merged #807.
